prerelease/action.yaml

```diff
@@ -0,0 +1,190 @@
+  GitHub Action to compute a prerelease version based on the latest release version and the number of commits since the
+  latest release.
+inputs:
+  checkout-repo:
+    required: false
+    description: Perform checkout as first step of action
+    default: "true"
+  create-prerelease:
+    required: false
+    description: Whether semantic-release should create a prerelease or do a dry run. This can be useful to set to true when a prerelease requires pushing artifacts semantic-release is in charge of generating
+    default: "true"
+  github-token:
+    required: true
+    description: GitHub token that can checkout the repository as well as create tags/releases against it. e.g. 'secrets.GITHUB_TOKEN'
+  extra-plugins:
+    required: false
+    description: Extra plugins for pre-install. You can also specify specifying version range for the extra plugins if you prefer.  Defaults to install @open-turo/semantic-release-config.
+    default: |
+      @open-turo/semantic-release-config
+outputs:
+  new-release-published:
+    description: Whether a new release was published
+    value: ${{ steps.prerelease.outputs.new-release-published }}
+  new-release-version:
+    description: Version of the new release
+    value: ${{ steps.prerelease.outputs.new-release-version }}
+  new-release-major-version:
+    description: Major version of the new release
+    value: ${{ steps.prerelease.outputs.new-release-major_version }}
+  pull-request-number:
+    description: Pull request number
+    value: ${{ steps.PR.outputs.number }}
+  run-url:
+    description: URL to the GHA run
+    value: ${{ steps.vars.outputs.run-url }}
+
+runs:
+  using: composite
+  steps:
+    - name: Dump GitHub context
+      env:
+        GITHUB_CONTEXT: ${{ toJSON(github) }}
+      shell: bash
+      run: |
+        echo "::group::Github Context"
+        echo "$GITHUB_CONTEXT"
+        echo "::endgroup::"
+
+    - name: Set vars
+      id: source-vars
+      shell: bash
+      env:
+        event_name: ${{ github.event_name }}
+        dispatch_client_payload_ref: ${{ github.event.client_payload.ref }}
+        dispatch_client_payload_sha: ${{ github.event.client_payload.sha }}
+        push_ref_name: ${{ github.ref_name }}
+        push_sha: ${{ github.sha }}
+        pull_request_ref_name: ${{ github.event.pull_request.head.ref }}
+        pull_request_sha: ${{ github.event.pull_request.head.sha }}
+      run: |
+        echo "event_name=$event_name"
+
+        if [ "$event_name" == "push" ]; then
+          branch=$push_ref_name
+          sha=$push_sha
+        elif [ "$event_name" == "repository_dispatch" ]; then
+          branch=$dispatch_client_payload_ref
+          sha=$dispatch_client_payload_sha
+        elif [ "$event_name" == "pull_request" ]; then
+          branch=$pull_request_ref_name
+          sha=$pull_request_sha
+        else
+          echo "::error::Unsupported event type '$event_name'"
+          exit 1
+        fi
+
+        echo "branch=$branch"
```

Comment:
why the double echo?

Reply:
One echos the value so you can see it. The other is echoed into the GitHub file. So you don't know what was set.